src/plugins/data/public/autocomplete/providers/query_syntax_provider.ts

```diff
@@ -17,56 +17,43 @@
  * under the License.
  */
 
-import { AutocompleteProviderRegister } from '.';
-import { IIndexPattern, IFieldType } from '../../common';
+import { IFieldType, IIndexPattern } from '../../../common/index_patterns';
 
-export type AutocompletePublicPluginSetup = Pick<
-  AutocompleteProviderRegister,
-  'addProvider' | 'getProvider'
->;
-export type AutocompletePublicPluginStart = Pick<AutocompleteProviderRegister, 'getProvider'>;
+export type QuerySyntaxSuggestionType =
+  | 'field'
+  | 'value'
+  | 'operator'
+  | 'conjunction'
+  | 'recentSearch';
 
-/** @public **/
-export type AutocompleteProvider = (args: {
-  config: {
-    get(configKey: string): any;
-  };
+export type QuerySyntaxProvider = (args: {
   indexPatterns: IIndexPattern[];
   boolFilter?: any;
-}) => GetSuggestions;
+}) => QuerySyntaxGetSuggestions;
 
-/** @public **/
-export type GetSuggestions = (args: {
+export type QuerySyntaxGetSuggestions = (args: {
   query: string;
```

Comment:
I think it would make sense to use the `Query` type here (to include the language)